text/logs/0000-logging-library-sdk.md

```diff
@@ -0,0 +1,251 @@
+  active Span.
+
+Appenders can be created in OpenTelemetry language libraries by OpenTelemetry
+maintainers, or by 3rd parties for any logging library that supports a similar
+extension mechanism. This specification recommends each OpenTelemetry language
+library to include out-of-the-box Appender implementation for at least one most
+popular logging library.
+
+### Logging to File
+
+One of the possible approaches to emit and collect logs that OpenTelemetry
+supports is via intermediary files. When configuring the LogEmitterProvider,
+OTLP File exporter should be used to write logs to a file or stdout in either
+OTLP JSON or OTLP Protobuf binary format.
+
+![Logging to File](images/otep0000/otlp-file.png)
+
+### Logging Directly to OTLP Network Destination
+
+The approach is the same as for logging to a file, except OTLP/gRPC or OTLP/HTTP
+exporter implementation is used.
+
+### Implicit Context Injection
+
+When Context is implicitly available (e.g. in Java) it may be fetched by the log
+library extension synchronously for every log record by calling the
+OpenTelemetry Context API and injecting the span context fields into the
+LogRecord before emitting it.
+
+Some log libraries have mechanisms specifically tailored for injecting
+contextual information into log records. An example of such a mechanism is Log4j
+MDC. When available such mechanisms may be the preferable place to fetch the
+span context and inject it into the log records, since it usually allows
+fetching of the context to work correctly even when log records are emitted
+asynchronously (which otherwise can result in the incorrect implicit context
+being fetched.
+
+### Explicit Context
+
+In languages where the Context must be provided explicitly (e.g. Go) the end
+user must capture the context and explicitly pass it to the logging subsystem in
+order for trace context to be recorded in Log records.
+
+Support for OpenTelemetry for logging libraries in these languages typically can
+be implemented in the form of logger wrappers that can capture the context once,
+when the span is created and then use the wrapped logger to execute log
+statements in a normal way. The wrapper will be responsible for injecting the
+captured context in the log records.
+
+This specification does not define how exactly it is achieved since the actual
+mechanism depends on the language and the particular logging library used. In
+any case the wrappers are expected to make use of the Trace Context API to get
+the current active span.
+
+See
+[an example](https://docs.google.com/document/d/15vR7D1x2tKd7u3zaTF0yH1WaHkUr2T4hhr7OyiZgmBg/edit#heading=h.4xuru5ljcups)
+of how it can be done for zap logging library for Go.
+
+### Custom LogExporter
+
+LogExporter implementations can be plugged into OpenTelemetry Logging Library to send logs via custom protocols.
+
+OTLP/gRPC, OTLP/HTTP, OTLP/File log exporters are provided with OpenTelemetry Logging Library out of the box.
```

Comment:
I wonder if some users would want to have OTLP/Console exporter as well (which could e.g. print JSON-formatted records)

Reply:
I too think we should provide OTLP/Console but, it would be OTLP/File(stdout) may we could add that wrapper too.

Reply:
Yes, the was the intent. OTLP/File can output to stdout as an option. We may also provide a wrapper to make it trivial.